### PR TITLE
[dagster-airlift] Fix materialization reporting order for tasks

### PR DESCRIPTION
**Pull request by dpeng817**
Fixes the materialization reporting order for asset keys mapped from airflow tasks to instead be topological.
Added a very basic test where a dependency between two tasks should always yield the downstream task being added after the upstream.

### CODE REVIEW

**Comment by dpeng817** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23326?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#23326** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23326?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#23324** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23324?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23300** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23300?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @dpeng817 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by schrockn** [comment]
Consider `toposort_flatten`

File: examples/experimental/dagster-airlift/dagster_airlift/airflow_utils.py [hunk 1]
```diff
@@ -304,3 +307,34 @@ def get_leaf_specs(specs: List[AssetSpec]) -> List[AssetSpec]:
         if key not in downstreams_per_key
     ]
     return leaf_specs
+
+
+def toposort_flatten(specs: List[AssetSpec]) -> List[AssetSpec]:
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/dagster_airlift/airflow_utils.py:312]: there's an existent toposort_flatten in the codebase. That's what I meant.

Comment (dpeng817) [reply examples/experimental/dagster-airlift/dagster_airlift/airflow_utils.py:312]: ah

File: examples/experimental/dagster-airlift/dagster_airlift/airflow_utils.py [hunk 2]
```diff
@@ -217,9 +218,12 @@ def airflow_dag_sensor(context: SensorEvaluationContext) -> SensorResult:
                                 "Link to Run": MarkdownMetadataValue(
                                     f"[View Run]({airflow_webserver_url}/dags/{dag_id}/grid?dag_run_id={dag_run['dag_run_id']}&tab=details)"
                                 ),
+                                "Creation Timestamp": TimestampMetadataValue(
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/dagster_airlift/airflow_utils.py:221]: Consider "Event Creation Timestamp"

**Review by schrockn — APPROVED** [review]

**Pull request merged**